src/server/service/kv.rs

```diff
@@ -677,6 +685,38 @@ impl<T: RaftStoreRouter + 'static> tikvpb_grpc::Tikv for Service<T> {
         ctx.spawn(future);
     }
 
+    fn raw_batch_get(
+        &self,
+        ctx: RpcContext,
+        mut req: RawBatchGetRequest,
+        sink: UnarySink<RawBatchGetResponse>,
+    ) {
+        const LABEL: &str = "raw_batch_get";
+        let timer = self.metrics.raw_batch_get.start_coarse_timer();
+
+        let keys = req.take_keys().into_iter().map(|x| x).collect();
+
```

Comment:
Please remove this blank line.

Reply:
Then which block do you want this line go to. From other methods, it looks like metrics related lines are grouped into one block and future is a separated block, whatever in between them, are grouped separately depends on their nature.

Reply:
@zhangjinpeng1987 Can you explain which block do you want this line go to. From other methods, it looks like metrics related lines are grouped into one block and future is a separated block, whatever in between them, are grouped separately depends on their nature.

Reply:
In this function, you get keys and then immediately use it, i think they should in the same block.

Reply:
I'm confused by all different kind of styles used in this file. For example, here uses a separate like to prepare keys.

 408     fn kv_batch_get(
 409         &self,
 410         ctx: RpcContext,
 411         mut req: BatchGetRequest,
 412         sink: UnarySink<BatchGetResponse>,
 413     ) {
 414         const LABEL: &str = "kv_batchget";
 415         let timer = self.metrics.kv_batchget.start_coarse_timer();
 416
 417         let keys = req.get_keys()
 418             .into_iter()
 419             .map(|x| Key::from_raw(x))
 420             .collect();
 421
 422         let future = self.storage
 423             .async_batch_get(req.take_context(), keys, req.get_version())
 424             .then(|v| {

Reply:
Sorry, didn't see the previous comment, we probably send out comments at the same time, and mine just came a little later to overlook yours. It's changed in this way now.